### PR TITLE
grading filter

### PR DESCRIPTION
**Pull request by aaronshiel**
- The pagination setup was already very flaky due to race conditions, tightened that up.
- Can filter by lesson name.
- Can filter by username.
- Page position and set filters all persist when returning from grading.
- Clicking Grade on a single lesson from the Lessons page brings you to the grade page with it filtered down to that lesson.
- Edit Single Lesson page has grade button that directs to Grade page with same filtering applied.

### CODE REVIEW

File: .github/workflows/test-and-publish.yml [hunk 1]
```diff
@@ -27,13 +27,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - run: docker-compose build
-      - run: docker-compose up --abort-on-container-exit --exit-code-from e2e
+      - run: docker compose build
```
Comment (dan100110) [inline .github/workflows/test-and-publish.yml:30]: might be wise to to switch this to the cypress action so it doesn't have to build docker images.  That would put it in line with RACR and Gameifai clients.

**Review by dan100110 — CHANGES_REQUESTED** [review]

File: .github/workflows/test-and-publish.yml (hunk 1 above)
Comment (aaronshiel) [reply .github/workflows/test-and-publish.yml:30]: What's the reasoning for no longer using docker images?

Comment (dan100110) [reply .github/workflows/test-and-publish.yml:30]: It simplifies the process of running the tests,  We no longer need to maintain an extra docker compose file and a separate docker build file for testing.

Comment (aaronshiel) [reply .github/workflows/test-and-publish.yml:30]: The question then would be why did we use them in the first place? I thought we wanted to use them to ensure a consistent working environment among other benefits

**Review by dan100110 — APPROVED** [review]

**Pull request merged**